packages/ripple-binary-codec/src/types/hash-128.ts

```diff
@@ -9,8 +9,25 @@ class Hash128 extends Hash {
   static readonly ZERO_128: Hash128 = new Hash128(Buffer.alloc(Hash128.width))
 
   constructor(bytes: Buffer) {
+    if (bytes && bytes.byteLength === 0) {
```

Comment:
I think this should happen in Hash similar to how it is done in xrpl-py.

Reply:
Added a PR for xrpl-py:
https://github.com/XRPLF/xrpl-py/pull/508